helm/vmss-prototype/README.md

```diff
@@ -126,8 +115,58 @@ The following Helm Chart values are exposed to configure a `vmss-prototype` rele
+  More documentation on node drain can be found [here in the `kubectl` documentation](https://kubernetes.io/docs/reference/generated/kubectl/kubectl-commands#drain).
+
 
 - `kamino.logLevel` (default value is `INFO`, options are `DEBUG`, `INFO`, `WARNING`, `ERROR`, and `CRITICAL`)
-  - e.g., `--set kamino.logLevel=VERBOSE`
+  - e.g., `--set kamino.logLevel=DEBUG`
   - Override the default if you wish to see more or less log output in the `vmss-prototype` pod.
+
+### Automatic node selection
+
+Kamino can follow some rules for automatic node selection.  These are the parameters for the helm chart (in addition to the above) to enable this.
+
+- `kamino.targetVMSS` (required to run in automatic mode)
+  - A value of `ALL` will automatically scan for all VMSS pools
+    - e.g., `--set kamino.targetVMSS=ALL`
+  - A value of the VMSS name.  This is the name of the VMSS without the node-specific identifier appended.  For example, `k8s-pool1-12345678-vmss` would be the VMSS that contains nodes such as `k8s-pool1-12345678-vmss000000`
+    - e.g., `--set kamino.targetVMSS=k8s-pool1-12345678-vmss`
+
+- `kamino.auto.lastPatchAnnotation` (required)
+  - This defines the name of the annotation on the nodes that holds an RFC3339 timestamp of when it last applied an OS patch.  If this annotation is missing, it is assumed the node has not applied an OS patch yet.
+    - e.g., `--set kamino.auto.lastPatchAnnotation=LatestOSPatch`
+
+- `kamino.auto.pendingRebootAnnotation` (required ??)
+  - This annotation on a node that indicates that the node is in need of reboot for some reason (pending OS patch or other servicing).  Nodes with this annotation are not considered as a viable candidate for making an image.
+    - e.g., `--set kamino.auto.pendingRebootAnnotation=RebootPending`
+  - Note that setting this to some annotation that never exists causes the system to not filter out any nodes due to this constraint.
+
+- `kamino.auto.minimumReadyTime` (default 1h)
+  - This is the minimum time a node must have been in constant "ready" state for it to be considered a potential candidate as the prototype.
+  - Values are logically in seconds unless a modifier is used.  For example:
+    - `--set kamino.auto.minimumReadyTime=3` --> 3 seconds
+    - `--set kamino.auto.minimumReadyTime=3s` --> 3 seconds
+    - `--set kamino.auto.minimumReadyTime=3m` --> 3 minutes
+    - `--set kamino.auto.minimumReadyTime=3h` --> 3 hours
+    - `--set kamino.auto.minimumReadyTime=3d` --> 3 days
+  - It is recommended to not have this value too low as this determines how long a node must have been operating before we consider it healthy enough to build future nodes from it.
+
+- `kamino.auto.minimumCandidates` (default 1)
+  - This is the minimum number of valid candidates before one is selected.  Must be at least 1 (since you can't selected from a candidate pool of size 0)

```

Comment:
Should we explain the value of setting a higher value here? I assume the idea is that we might want a broader depth of operational "node health" (more candidates means you're more likely to be ready for an update) before updating the cluster?

Reply:
I thought that is what I did just below the example.